engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2260,6 +2260,29 @@ public void shutdownAllContainers(String message)
     }
   }
 
+  /**
+   * Send STOP_OPERATOR command to all input operator to initiate shutdown of a DAG.
+   *
+   * @param message
+   */
+  public void stopInputOperators(String message)
```

Comment:
We can change the *shutdown* variable in the heartbeat to Enum to represent different ways of shutting the containers and operators. Idea is that all the shutdown logic is contained in one place.